lua/telescope/actions/init.lua

```diff
@@ -120,7 +120,7 @@ function actions._goto_file_selection(prompt_bufnr, command)
         vim.cmd(string.format(":tab sb %d", entry_bufnr))
       end
     else
-      filename = path:new(vim.fn.fnameescape(filename)):normalize()
+      filename = path:new(vim.fn.fnameescape(filename)):normalize(vim.fn.getcwd())
```

Comment:
Can we check if we still need `vim.fn.fnameescape` when doing path:new? @elianiva 

Also I kinda wanna move all `vim.fn.getcwd()` to `vim.loop.cwd()` to reduce our use of vimL

Reply:
was about to try this but I got this error.
```
...packer/start/telescope.nvim/lua/telescope/make_entry.lua:122: attempt to call method 'make_relative' (a nil value)
```

Reply:
Oh yeah requires a plenary PR : https://github.com/nvim-lua/plenary.nvim/pull/60

I am not sure if its done already 😆 but thanks for testing

Reply:
Ah, so that's that I was missing :laughing:
Unfortunately we still need to escape it. You think it's a good idea to escape it ourself instead of relying on `fnameescape`?
My issue isn't that common and the only thing that's causing it is because `$` conflict with bash/zsh variable naming.

Reply:
Could you try this path for plenary on top of that PR:
```diff
diff --git a/lua/plenary/path.lua b/lua/plenary/path.lua
index 33372f9..c42c97e 100644
--- a/lua/plenary/path.lua
+++ b/lua/plenary/path.lua
@@ -200,14 +200,15 @@ end
 
 function Path:make_relative(cwd)
   cwd = F.if_nil(cwd, self._cwd, cwd)
-  if self.filename:sub(1, #cwd) == cwd  then
+  local filename = self:absolute() or self.filename
+  if filename:sub(1, #cwd) == cwd  then
     local offset =  0
     -- if  cwd does ends in the os separator, we need to take it off
     if cwd:sub(#cwd, #cwd) ~= path.separator then
       offset = 1
     end
 
-    self.filename = self.filename:sub(#cwd + 1 + offset, #self.filename)
+    self.filename = filename:sub(#cwd + 1 + offset, #filename)
   end
 
   return self.filename
@@ -217,7 +218,8 @@ function Path:normalize(cwd)
   cwd = F.if_nil(cwd, self._cwd, cwd)
   self:make_relative(cwd)
   -- Substitute home directory w/ "~"
-  self.filename = self.filename:gsub("^" .. path.home, '~', 1)
+  local filename = self:absolute() or self.filename
+  self.filename = filename:gsub("^" .. path.home, '~', 1)
   -- Remove double path seps, it's annoying
   self.filename = self.filename:gsub(path.sep .. path.sep, path.sep)
   print(self.filename)
@@ -247,7 +249,7 @@ local shorten = (function()
 end)()
 
 function Path:shorten()
-  return shorten(self.filename)
+  return shorten(self:absolute() or self.filename)
 end
 
 function Path:mkdir(opts)
```

Reply:
Ok it's sorta works. I can open the file but it's not a valid file I think. Its content is different from what I have and I can't the file.

Reply:
Hmm okay. I think `Path:new` should handle the escaping better than vimL 🤣 . I will take a look at it later today. Shouldn't be to hard to create a file with `$` and do some testing 😆 Or maybe @anott03 wants to look at it :)

Still thanks for testing :)

Reply:
no problem :)